server/app.js

```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
 
 http.listen(port,function(){
     log.info('Server listening through port %s', port);
-});
\ No newline at end of file
+});
+
+http.on('err', function(err){
+   // handle the error safely
+    console.log(err)
+});
+
+process.on('uncaughtException', function(err) {
+  console.log('Caught exception: ' + err);
+});
+
+
+var state = ["NAVIGATING_SIDEWALK", "LOOKING_FOR_PATH" ,"FOUND_PATH_TO_HOUSE", "NAVIGATING_PATH" ,"REACHED_DOOR" , "DELIVER_PACKAGE" ]
+var current_state;
+var repeat;
+var state_repeat;
+call_number_recognition();
+call_video_stream();
+//init_state_machine();
+
+
+function init_state_machine() {
+    // Initial state Navigating Sidewalk
+
+                client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+                    console.log('Connected to 192.168.43.250'); 
+                    console.log('Local client sending:', 0);
+                    buf = new Buffer([14, 0x4, 0xFF]); // 0x4B is temp value 75
+                    client.write(buf);
+                    client.destroy();
+                    client = new net.Socket();
+                    client.on('err', function(err){
+                        // handle the error safely
+                        console.log(err)
+                    });
+                });
+
+    repeat = setInterval(navigating_sidewalk, 500);
+}
+
+function restart_state_machine(){
+    stop_state_machine();
+    init_state_machine();
+}
+
+function stop_state_machine(){
+    clearInterval(repeat);
+    clearInterval(state_repeat);
+}
+
+function call_video_stream() {
+    console.log("Calling video stream script.\n");
+    var pyshell = new PythonShell('../sidewalk/python/video_stream.py')
+}
+
+function navigating_sidewalk(){
+    //Read
+    current_state = state[0];
+
+    console.log("navigating sidewalk");
+
+    var file = 'front.json';
+    jsonfile.readFile(file, function(err, obj) {
+        //console.log(obj.angle);
+
+        if(obj.angle >= 85 && obj.angle <= 95 ){
+                client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
```

Comment:
These blocks can be a function for less repetition.

Reply:
Kinda embarassing I didn't do this earlier `:^) thanks